woocommerce/woocommerce-ios
# Adding new style for unit input cell that places the price input on the left

closes https://github.com/woocommerce/woocommerce-ios/issues/6563

### Description
This PR adds a new style for the `UnitInputTableViewCell` that places the input field in the left of the cell and does not support any title text, to match the mockups for the price editing screen in #6242. The style can be selected using a value of the `UnitInputViewModel.Style` enum. The new style is the `.secondary` and the `.primary` is the original.
The default (`.primary`) style has a layout with the price field on the right, more specifically starting from the left it has  a title label and then a stackView that contains the input field and a label for the unit(e.g. $). In order to place the input field on the left for the secondary style we set an empty title, then we set to zero the spacing between the left title label and the stackview (`inputAndUnitStackViewTotitleLabel `), and finally we change the content hugging priority of the field to defaultLow in order for this field (instead of the title label) to stretch and take all the available space. Also since in the secondary style the input field is on the left we adjust the `textAligment` of the input field to be left for RTL and right otherwise.

This PR adds an extension of the `UnitInputViewModel` with a method for creating viewModels with the secondary style needed to configure the cell for the bulk price update screen.

### Testing instructions
Testing instructions

1. Goto the products tab
2. Select a product with variations
3. Go to the list of variations
4. Tap on the "more" button in the navigation bar
5. Select the "Bulk update"
6. The bulk update screen should appear, select the regular price option
7. The price setting screen should appear8. 
8. The price input field should be on the left, with a place holder "0.00", and no extra title

### Screenshots

New input cell layout with the price on the left:

Existing `.primary` style:

![Screenshot 2022-03-14 at 8 57 59 AM](https://user-images.githubusercontent.com/96764631/158130512-a42fd3e0-f4d0-46b5-9b32-9a147ccc0447.png)


'.secondary' style:

![Simulator Screen Shot - iPod touch (7th generation) - 2022-03-14 at 07 30 13](https://user-images.githubusercontent.com/96764631/158112808-709df74c-9a62-41c0-acb1-e83bbcab6bdf.png)


![Simulator Screen Shot - iPod touch (7th generation) - 2022-03-14 at 07 30 16](https://user-images.githubusercontent.com/96764631/158112789-b9d1b643-1539-48fe-9d3d-d1714a5e3841.png)

---
